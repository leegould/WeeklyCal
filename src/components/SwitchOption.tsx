import React from 'react';
import { Switch, StyleSheet } from 'react-native';
import { ListItem } from 'react-native-elements';

type Props = {
    // key: string,
    title: string,
    subtitle: string,
    value: boolean,
    onValueChange: Function,
    trackColor?: string,
}

export default class Option extends React.PureComponent<Props> {
    render() {
        const trackColor = this.props.trackColor ? this.props.trackColor : '#C2272D';

        return(
            <ListItem
                key={this.props.title.replace(' ', '')}
                title={this.props.title}
                subtitle={this.props.subtitle}
                rightAvatar={
                    <Switch
                        onValueChange={() => this.props.onValueChange()}
                        value={this.props.value}
                        trackColor={{true: trackColor, false: trackColor}}
                        style={styles.switchInput}
                    />
                }
                containerStyle={{ backgroundColor: 'lightgray', borderBottomWidth: 1, borderBottomColor: 'gray' }}
                titleStyle={{ color: 'white' }}
                subtitleStyle={{ color: 'white', fontSize: 10, marginVertical: 5 }}
            />
        );
    }
}

const styles = StyleSheet.create({
    switchInput: {
        transform: [{ scaleX: .8 }, { scaleY: .8 }]
    },
});