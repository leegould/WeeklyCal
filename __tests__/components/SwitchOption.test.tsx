import React from 'react';
import renderer from 'react-test-renderer';
import SwitchOption from '../../src/components/SwitchOption';

jest.useFakeTimers();

test('renders correctly with expected args', () => {
    const props = {
        title: 'string',
        subtitle: 'string',
        value: false,
        onValueChange: () => {},
        trackColor: 'string',
    }

    const tree = renderer.create(
        <SwitchOption {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
