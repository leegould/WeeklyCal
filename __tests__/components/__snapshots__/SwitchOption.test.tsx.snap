// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with expected args 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "borderBottomColor": "gray",
        "borderBottomWidth": 1,
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "white",
            "fontSize": 17,
          }
        }
        testID="listItemTitle"
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        string
      </Text>
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "white",
            "fontSize": 10,
            "marginVertical": 5,
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        string
      </Text>
    </View>
    <View
      width={16}
    />
    <RCTSwitch
      accessibilityRole="button"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="string"
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Object {
            "transform": Array [
              Object {
                "scaleX": 0.8,
              },
              Object {
                "scaleY": 0.8,
              },
            ],
          },
        ]
      }
      tintColor="string"
      value={false}
    />
  </View>
</View>
`;
